asmcomp/cmm.mli

```diff
@@ -134,10 +134,10 @@ and expression =
   | Clet of Ident.t * expression * expression
   | Cassign of Ident.t * expression
   | Ctuple of expression list
-  | Cop of operation * expression list
+  | Cop of operation * expression list * Debuginfo.t
   | Csequence of expression * expression
   | Cifthenelse of expression * expression * expression
-  | Cswitch of expression * int array * expression array
+  | Cswitch of expression * int array * expression array * Debuginfo.t
```

Comment:
Well, this probably comes from taking an SSA-style view on things, but to me `Cassign` is just the same as `Clet` it doesn't actually do anything at runtime it tells the compiler to give an existing thing a particular name. To me, when you turn a local reference into a `Cassign` you are removing the mutable thing and replacing it by just passing the most recent value around to the appropriate locations -- the act of mutating has been taken out of the semantics.

If/when we switch to SSA-style that would be more obviously true, rather than just my interpretation of things.